src/interface/meshblock_data.hpp

```diff
@@ -134,7 +143,18 @@ class MeshBlockData {
       }
     } else {
       for (const auto &v : vars) {
-        add_var(src->GetVarPtr(v));
+        auto var = src->GetVarPtr(v);
+        add_var(var);
+        // Add the associated flux as well if not explicitly
+        // asked for
+        if (var->IsSet(Metadata::WithFluxes)) {
+          auto flx_name = var->metadata().GetFluxName();
+          bool found = false;
+          for (const auto &v2 : vars) {
+            if (src->GetVarPtr(v2)->label() == flx_name) found = true;
+          }
+          if (!found) add_var(src->GetVarPtr(flx_name));
```

Comment:
Do we need to check if the found variable has certain properties (required and incompatible -- like `OneCopy` or similar)?

Reply:
I don't think that we want to do that, at least for backward compatibility. If a variable is in a `MeshBlockData` object, it's flux should be available there as well. At least that was my thinking.